src/libasr/codegen/llvm_utils.cpp

```diff
@@ -4659,10 +4659,26 @@ namespace LCompilers {
 
```

Comment:
Can you please update the equivalent C++ code above, to make it clear how the logic works?

Reply:
Sure, if we are going ahead with the change I will update the logic as well.
For understanding the logic is below - 
```
if(pos<0){
      if(abs(pos)<= end_point){
            adjusted_pos = endpoint + pos
    }
}
else{
   adjusted_pos = pos
}
if(adjusted_pos <0 || adjusted_pos>=end_point) return IndexError
```